### PR TITLE
refactor: migrate dropdown validator to ts

### PR DESCRIPTION
**Pull request by tshuli**
_Blocked by table validator migration_

## Problem
Part of #7

## Solution

**Improvements**

- Migrates dropdown validation to Typescript

## Tests
- [ ] Create two dropdown fields. Hide one of them behind logic.
- [ ] Check that validation on both visible and hidden dropdown fields work when submitting programatically
- Response is rejected if it is not one of the dropdown options
- [ ] Create Myinfo field with dropdown (e.g. nationality). Check that form can be submitted normally.

### CODE REVIEW

**Review by liangyuanruo — CHANGES_REQUESTED** [review]
> Blocked by table validator migration

Could we migrate validation for dropdown fields, but leave the table validator to rely on the previous dropdown implementation until we migrate that?

**Comment by tshuli** [comment]
@liangyuanruo restored the previous dropdown implementation (for TableValidator), for your review

**Comment by tshuli** [comment]
debugging failed e2e tests for myinfo fields

**Review by liangyuanruo — CHANGES_REQUESTED** [review]
to address failing tests before approving

File: src/app/utils/field-validation/validators/dropdownValidator.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  const { myInfo } = dropdownField
+  if (myInfo && myInfo.attr) {
+    const [myInfoField] = myInfoTypes.filter(
+      (type) => type.name === myInfo.attr,
+    )
+    const { fieldOptions } = myInfoField
+    if (fieldOptions) dropdownField.fieldOptions = fieldOptions
+  }
+
+  const { fieldOptions } = dropdownField
```
Comment (liangyuanruo) [inline src/app/utils/field-validation/validators/dropdownValidator.ts:32]: suggest extracting the finding of the correct MyInfo fieldOptions into a separate utility function for reuse, then you can express the logic more succinctly with a ternary op:

```typescript
const validOptions = myInfo?.attr ? getMyInfoOptions(myInfo.attr) : fieldOptions
return isOneOfOptions(validOptions, answer)
  ? right(response)
  : left(`DropdownValidator:\t answer is not a valid dropdown option`)
```

**Review by liangyuanruo — APPROVED** [review]

File: src/app/utils/field-validation/validators/dropdownValidator.ts (hunk 1 above)
Comment (liangyuanruo) [reply src/app/utils/field-validation/validators/dropdownValidator.ts:32]: more generally, we should always find the correct level of abstraction to achieve clean code. in this example we want to switch between `fieldOptions` depending on whether it is a MyInfo field or not. we don't want a reader of the code to delve into implementation details to find out.

**Pull request merged**